### PR TITLE
Changes annotation routes to attributes

### PR DESCRIPTION
**Pull request by SimonRothUCF**
Resolves [#774](https://github.com/ucfopen/UDOIT/issues/774)

### CODE REVIEW

**Comment by rob-3** [comment]
If you have any trouble fixing the conflicts, feel free to ping me. Also, just a style thing, but for commit messages typically the style is a phrase that completes "This commit will..." in a grammatically correct way. Eg "Fix error caused by rebase" instead of "Fixed error caused by rebase" or "Fixes error caused by rebase"

**Comment by SimonRothUCF** [comment]
Conflicts should be fixed. And I will start styling my commits that way, thanks for the tip. As long as it looks good to you, it should be ready to merge.

File: src/Controller/AuthController.php [hunk 1]
```diff
@@ -22,16 +22,14 @@ class AuthController extends AbstractController
...
     {
         $this->doctrine = $doctrine;
     }
 
-    /**
-     * @Route("/authorize", name="authorize")
-     */
     public function authorize(
         Request $request,
         SessionService $sessionService,
```
Comment (rob-3) [inline src/Controller/AuthController.php:35]: This doesn't look right; I think this attribute got a little lost in the merge.

File: src/Controller/DashboardController.php [hunk 2]
```diff
@@ -24,16 +24,14 @@ class DashboardController extends AbstractController
...
     private ManagerRegistry $doctrine;
 
     public function __construct(ManagerRegistry $doctrine)
     {
         $this->doctrine = $doctrine;
     }
 
-    /**
-     * @Route("/dashboard", name="dashboard")
-     */
```
Comment (rob-3) [inline src/Controller/DashboardController.php:36]: Same thing here; I think this attribute got moved by the rebase/merge.

File: src/Controller/FileItemsController.php [hunk 3]
```diff
@@ -13,16 +13,14 @@
...
     {
         $this->doctrine = $doctrine;
     }
 
-    /**
-     * @Route("/api/files/{file}/review", name="review_file")
-     */
     public function reviewFile(FileItem $file, Request $request, UtilityService $util)
     {
         $apiResponse = new ApiResponse();
```
Comment (rob-3) [inline src/Controller/FileItemsController.php:26]: Also I think this got lost

File: src/Controller/IssuesController.php [hunk 4]
```diff
@@ -14,19 +14,15 @@
...
 
-    /**
-     * Save change to issue HTML to LMS
-     * 
-     * @Route("/api/issues/{issue}/save", name="save_issue")
-     * @param Issue $issue
-     */
     public function saveIssue(
         Request $request, 
         LmsPostService $lmsPost,
```
Comment (rob-3) [inline src/Controller/IssuesController.php:28]: Another lost attribute

File: src/Controller/LtiController.php [hunk 5]
```diff
@@ -27,16 +27,14 @@ class LtiController extends AbstractController
...
         $this->doctrine = $doctrine;
     }
-
-    /**
-     * @Route("/lti/authorize", name="lti_authorize")
-     */
+  
     public function ltiAuthorize(
         Request $request,
         SessionService $sessionService,
```
Comment (rob-3) [inline src/Controller/LtiController.php:40]: Another lost attribute

File: src/Controller/ReportsController.php [hunk 6]
```diff
@@ -18,22 +18,18 @@ class ReportsController extends ApiController
...
     }
 
-    /**
-     * @Route("/api/courses/{course}/reports", methods={"GET"}, name="get_reports")
-     * @param $courseId
-     * @return JsonResponse
-     */
     public function getAllReports(
         UtilityService $util,
         Course $course
```
Comment (rob-3) [inline src/Controller/ReportsController.php:31]: Another lost attribute

File: src/Controller/ReportsController.php [hunk 7]
```diff
@@ -112,16 +103,12 @@ public function getLatestReport(Course $course)
...
-     * @param Course $course
-     * @return \Symfony\Component\HttpFoundation\Response
-     */
+    #[Route('/download/courses/{course}/reports/pdf', methods: ['GET'], name: 'get_report_pdf')]
     public function getPdfReport(
         Request $request,
         UtilityService $util,
         Course $course
-    ) {
+    ): \Symfony\Component\HttpFoundation\Response {
```
Comment (rob-3) [inline src/Controller/ReportsController.php:111]: I would suggest importing this at the top so you can just use `Response` here

File: src/Controller/UserController.php [hunk 8]
```diff
@@ -11,16 +11,14 @@
...
     {
         $this->doctrine = $doctrine;
     }
-
-    /**
-     * @Route("/api/users/{user}", name="user_put",  methods={"PUT"})
-     */
+  
     public function update(User $user, Request $request): JsonResponse
     {
```
Comment (rob-3) [inline src/Controller/UserController.php:23]: Another stranded attribute

**Review by rob-3 — CHANGES_REQUESTED** [review]
I think a couple of the attributes got misplaced, but aside from that and a few other minor things, this is looking pretty good.

File: src/Controller/ReportsController.php (hunk 7 above)
Comment (SimonRothUCF) [reply src/Controller/ReportsController.php:111]: Yes, you're right, thank you for catching those. I believe those got jumbled while I was fixing merge conflicts.

**Comment by SimonRothUCF** [comment]
Routes have been un-jumbled, and I made that change to `Response` in `ReportsController.php` as well. Anything else of note, or is it good to go?

File: src/Controller/AdminController.php [hunk 9]
```diff
@@ -280,12 +269,8 @@ public function getAdminLatestReport(Course $course, UtilityService $util, LmsAp
...
     }
 
-    /**
-     * @Route("/api/admin/users", methods={"GET"}, name="admin_users")
-     * 
-     * @return JsonResponse
-     */
-    public function getUsers(UserRepository $userRepo)
+    #[Route('/api/admin/users', methods: ['GET'], name: 'admin_users')]
+    public function getUsers(UserRepository $userRepo, ReportRepository $reportRepo): JsonResponse
```
Comment (rob-3) [inline src/Controller/AdminController.php:273]: The `ReportRepository` looks like it got added here when it didn't need to be

File: src/Controller/IssuesController.php [hunk 10]
```diff
@@ -103,14 +99,9 @@ public function saveIssue(
...
-     * Mark issue as resolved/reviewed
-     * 
-     * @Route("/api/issues/{issue}/resolve", methods={"POST","GET"}, name="resolve_issue")
-     * @param Issue $issue
-     * @return \Symfony\Component\HttpFoundation\Response
-     */
-    public function markAsReviewed(Request $request, LmsPostService $lmsPost, UtilityService $util, Issue $issue)
+    // Mark issue as resolved/reviewed
+    #[Route('/api/issues/{issue}/resolve', methods: ['POST','GET'], name: 'resolve_issue')]
+    public function markAsReviewed(Request $request, LmsPostService $lmsPost, UtilityService $util, Issue $issue): \Symfony\Component\HttpFoundation\Response
```
Comment (rob-3) [inline src/Controller/IssuesController.php:104]: I think this should just return `JsonResponse` like the other routes

File: src/Controller/UserController.php [hunk 11]
```diff
@@ -17,10 +17,8 @@ public function __construct(ManagerRegistry $doctrine)
...
         $this->doctrine = $doctrine;
     }
-
-    /**
-     * @Route("/api/users/{user}", name="user_put",  methods={"PUT"})
-     */
+
```
Comment (rob-3) [inline src/Controller/UserController.php:20]: There's extra spaces here; the newline should not have any space characters in it

**Review by rob-3 — CHANGES_REQUESTED** [review]

**Comment by SimonRothUCF** [comment]
Changes made. The last commit can be ignored, I thought it was causing problems but it was an issue on my end.

**Pull request merged**